packages/react-components/src/shared/src/component-types.ts

```diff
@@ -0,0 +1,28 @@
+import { ElementType, ForwardedRef, PropsWithChildren, ReactNode } from "react";
```

Comment:
Is the file name alright? 

The plan is to have this file be refered by all the components to import redundant types